[blaukc] iP

blaukc opened the pull request:
# Dude
- Does things
- Does more things

Here is a numbered list
1. The first thing it does is ```inline code```
2. The second thing _it_ **does** [not do](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

Here's some code:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
TODO:
- [ ] task not done
- [x] task done 
> 🌵

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

blaukc committed:
Change name to Dude

blaukc committed:
Split greeting and goodbye into their own functions

blaukc committed:
Add echo loop

blaukc committed:
Fix bug with ==

blaukc committed:
Make methods non static, add list function

blaukc committed:
Add Task.java

blaukc committed:
Add mark and unmark function

blaukc committed:
Add Todo, Deadline, Event subclass

blaukc committed:
Add the subclasses to Dude

blaukc committed:
Remove while true loop

blaukc committed:
Add tests

blaukc committed:
Fix Event showing as [D]

blaukc committed:
Add error handling

blaukc committed:
Add delete function

blaukc committed:
Add parameter type enum

blaukc committed:
Add Storage class

blaukc committed:
Add LocalDate

blaukc committed:
Add Parser and Ui

blaukc committed:
Add Parser.parse

blaukc committed:
Add TaskList

blaukc committed:
Refactor packages

blaukc committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2324S2/ip

blaukc committed:
Update main file to dude

blaukc committed:
Add ParserTest and StorageTest

blaukc committed:
Added javadocs

blaukc committed:
Fix coding standard (i hope)

blaukc committed:
Fix conflicts

blaukc committed:
Add find command

blaukc committed:
Fix conflicts

blaukc committed:
Fix checkstyle issues in project

blaukc committed:
Make arbitrary commit for testing

blaukc committed:
Make arbitrary commit for testing

blaukc committed:
Make arbitrary commit for testing

blaukc committed:
Make arbitrary commit for testing

blaukc committed:
Create basic structure for GUI

blaukc committed:
Add dialog boxes to gui

blaukc committed:
Add response for Dude into the GUI

blaukc committed:
Add FXML and style changes

tejas-1405 commented on src/main/java/dude/task/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Override
+    public String getStorageString() {
```
correct naming convention for method!

tejas-1405 commented on src/main/java/dude/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+ * Task parent class.
+ */
+public abstract class Task {
+    private boolean done = false;
```
Could rename the variable to something like isDone

tejas-1405 commented on src/main/java/dude/task/Todo.java:
```diff
@@ -0,0 +1,33 @@
+     * Creates Todo object.
+     * @param description Todo description.
+     * @param done  Whether todo is done.
+     */
```
Good use of JavaDocs

tejas-1405 commented on src/main/java/dude/Parser.java:
```diff
@@ -0,0 +1,137 @@
+     * @param str String input of a date.
+     * @return Whether input is in the correct date format.
+     */
+    public static boolean isDate(String str) {
```
Good naming convention for methods!

tejas-1405 reviewed (commented):
Overall looks good dude

yeoshuheng commented on src/main/java/dude/DialogBox.java:
```diff
@@ -0,0 +1,69 @@
+package dude;
```
Maybe the codebase would be better structured if you extracted all the javaFX components to a separate package.

yeoshuheng commented on src/main/java/dude/Parser.java:
```diff
@@ -0,0 +1,137 @@
+     * @param str String input of a number.
+     * @return Whether input is numeric.
+     */
+    public static boolean isNumeric(String str) {
```
Great work! Very elegant way of handling input issues in the parser.

yeoshuheng commented on src/main/java/dude/Storage.java:
```diff
@@ -0,0 +1,147 @@
+     * Class constructor.
+     */
+    public Storage() {
+        boolean success = createStorageIfNotExists();
```
Instead of just success, would it be better to attribute some sort of name to it?
But this could just be a nit :)

```suggestion
        boolean storageCreationIsSuccesful = createStorageIfNotExists();
```

yeoshuheng commented on src/main/java/dude/Storage.java:
```diff
@@ -0,0 +1,147 @@
+        new File(Storage.directoryPath).mkdirs();
+        File storageFile = new File(this.filePath);
+        if (!storageFile.exists()) {
+            System.out.println("dude.Storage file not found. Creating storage file...");
```
Would it be better to throw these lines up to the Ui to reflect to the user the state of the program too?

yeoshuheng commented on src/test/java/dude/ParserTest.java:
```diff
@@ -0,0 +1,31 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ParserTest {
```
Very thorough testing, great work!

yeoshuheng commented on src/main/java/dude/Storage.java:
```diff
@@ -0,0 +1,147 @@
+     */
+    public boolean createRows(ArrayList<Task> tasks) {
+        boolean success = this.clearStorage();
+        if (!success) {
```
I love how you are throwing success / failure upwards for functions like this, however, upon failure, would it be better to throw a error from the lower components and handle it instead of just solely passing booleans?

yeoshuheng reviewed (commented):
Great work!

I find your code well structured and well designed to handle the iP requirements.
I also love the thought you have put into handling various types of abstractions, going above the requirements set upon us.

blaukc committed:
Add assertions to Dude

Current code does not implement assertions.
Need to add assertions to check for errors made by developers.

Adding assertions for some switch statements and conditional statements.
Also adding assertions for some assumptions like the createRows
functions and getResponse.

blaukc committed:
Merge pull request #1 from blaukc/branch-A-Assertions

Add assertions to Dude

blaukc committed:
Remove commented out code

Getting rid of unused code the moment it becomes redundant.
This is to help to prevent clutter.

blaukc committed:
Refactor command parameters and command parameter types

Refactoring to make the Dude command parse switch statement more
readable.
Use a static Map to store all the constants at the top of the class.

Also removed some unused imports.

blaukc committed:
Change done to isDone

blaukc committed:
Merge pull request #2 from blaukc/branch-A-CodeQuality

Branch a code quality

blaukc committed:
Add yml file for CI

blaukc committed:
Fix style issues in Dude

blaukc committed:
Remove the last 3 tests because Dude no longer uses GUI

blaukc committed:
Fix checkstyle for Dude tests

blaukc committed:
Add view schedule command to Dude

blaukc committed:
Fix bug with full text not showing

blaukc committed:
Add new UI graphic for Dude

blaukc committed:
Update README with image and command information

blaukc committed:
Update title of README to Dude
